src/plugins/data/common/data_views/types.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ErrorToastOptions, ToastInputFields } from 'src/core/public/notifications';
+// eslint-disable-next-line
+import type { SavedObject } from 'src/core/server';
+import { FieldFormat, DataViewField, OSD_FIELD_TYPES } from '..';
+import { SerializedFieldFormat } from '../../../expressions/common';
+import { IDataViewFieldType } from './fields';
+
+export type DataViewFieldFormatMap = Record<string, SerializedFieldFormat>;
+
+export interface IDataView {
```

Comment:
Shouldn't this extend the index pattern interface? How are we ensuring interop? I thought part of the goal was that existing code could be swapped to Datasets by just changing types.